packages/mermaid/src/diagrams/common/common.ts

```diff
@@ -178,16 +178,18 @@ export const getMin = function (...values: number[]): number {
  * @returns The converted string
  */
 export const parseGenericTypes = function (text: string): string {
+  if (!text) {
+    return '';
+  }
+
   let cleanedText = text;
 
   if (text.split('~').length - 1 >= 2) {
     let newCleanedText = cleanedText;
 
-    // use a do...while loop instead of replaceAll to detect recursion
-    // e.g. Array~Array~T~~
     do {
       cleanedText = newCleanedText;
-      newCleanedText = cleanedText.replace(/~([^\s,:;]+)~/, '<$1>');
+      newCleanedText = cleanedText.replace(/~([^:;]+)~/, '<$1>');
     } while (newCleanedText != cleanedText);
 
     return parseGenericTypes(newCleanedText);
```

Comment:
This line should be removed. If there is a bug in the regex that fails to remove all the tildes, you will recurse infinitely and never reach the base-case (hence the "Maximum call Stack size exceeded" errors reported in #4431)

Reply:
@jakraft @sidharthv96 @aloisklink  The more and more I think on this, and try different approaches, the more I think that we should make the "breaking" change to ensure there is a clear distinction between 'start' and 'end' tokens.

I have tried doing this multiple ways, and still get hit by the max call stack size error

Reply:
@jgreywolf I don't think it necessarily requires a breaking change. Backward compatibility with the existing "~" format can be retained, but a new syntax is required to add support for multiple generics. The easiest approach I can see is to add support for angle brackets directly ("<>"). This would be fairly trivial to do, it would just require replacing any instances of `<` and `>` with `&lt;` and `&gt;` respectively. This can be done at the top of the parseGenericTypes function before handling the ~ logic.

Is there any reason the recursive call can't be removed regardless? I can't really see a reason to keep it in there.

Reply:
I think I have a solution

Reply:
Please see updated code and resolve if acceptable ;)  It does get rid of the max call size issue, and accounts for a couple of edge cases - if more "edge" cases develop this entire solution will have to be revisited.